### PR TITLE
chart: fix labels of components `schedulerEstimator` and `search`

### PR DESCRIPTION
**Pull request by calvin0327**
Signed-off-by: calvin0327 <wen.chen@daocloud.io>

**What type of PR is this?**
/kind bug

**What this PR does / why we need it**:
when install components `schedulerEstimator` and `search`, show follow err message:
```shell
helm install karmada -n karmada-system --set components={"search"} --set installMode=component  --create-namespace --dependency-update ./charts/karmada --debug --dry-run
```
62C0D287-3753-4202-9989-487F8E044E8B

```shell
helm install karmada -n karmada-system --set components={"schedulerEstimator"} --set installMode=component  --create-namespace --dependency-update ./charts/karmada --debug --dry-run
```
0F1B0053-2226-4907-9153-F2C59183E790

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:
please correct me if not

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

**Comment by calvin0327** [comment]
@Poor12 PTAL

**Comment by calvin0327** [comment]
> I have tested this issue. It does happen. But I have a question. In the early version, it should not happen. Do you know how the problem come about？

you can exec following cmd：
```shell
helm install karmada -n karmada-system --set components={"search"} --set installMode=component  --create-namespace --dependency-update ./charts/karmada --debug --dry-run
```

and

```shell
helm install karmada -n karmada-system --set components={"schedulerEstimator"} --set installMode=component  --create-namespace --dependency-update ./charts/karmada --debug --dry-run
```

File: charts/karmada/values.yaml [hunk 1]
```diff
@@ -768,7 +779,6 @@ search:
   ## @param search.labels labels of the search deployment
   labels:
     app: karmada-search
-    apiserver: "true"
```
Comment (Poor12) [inline charts/karmada/values.yaml:771]: I think just keep this change is ok. The other changes don't seem to make much sense. The reason for the failure of the scheduler Estimator is that the clusterName is not filled in.

Comment (calvin0327) [reply charts/karmada/values.yaml:771]: I'm sorry, my ide plugin have a bit problem. I will modify agin.

Comment (calvin0327) [reply charts/karmada/values.yaml:771]: @Poor12 I have a test, the failure reason of scheduler Estimator due to clusterName.
but the labels is necessary to scheduler Estimator.

File: charts/karmada/values.yaml [hunk 2]
```diff
@@ -56,15 +56,6 @@ kubectl:
     ##
     pullPolicy: IfNotPresent
 
-## pre-install job config
```
Comment (Poor12) [inline charts/karmada/values.yaml:59]: There three lines are due to https://github.com/karmada-io/karmada/pull/2348#discussion_r943093439.

Comment (calvin0327) [reply charts/karmada/values.yaml:59]: ok

Comment (calvin0327) [reply charts/karmada/values.yaml]: done

**Review by RainbowMango — APPROVED** [review]
/approve

**Pull request merged**